### ISSUE TITLE
New rule proposal: avoid several semi-colons in a row. 

### ISSUE THREAD

**Issue by tunetheweb** [sqlfluff/sqlfluff#2653]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

Multiple semi-colons are valid SQL and we even have a test case for this:

```sql
-- It's possible to have multiple semicolons between statements.
SELECT foo FROM bar;;
SELECT foo FROM bar;
;
;
SELECT foo FROM bar;
```

I think we should have a rule (or perhaps an enhancement to L052?) that removes these and replaces them with a single semi-colon.

### Use case

Above SQL should be changed to this:

```sql
SELECT foo FROM bar;
SELECT foo FROM bar;
SELECT foo FROM bar;
```

Where extraneous semi-colons (and whitespace surrounding them?) should be removed.

### Dialect

All

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
See also #2669, another open issue for L052. I strongly suggest we refactor L052 to simplify the code **before** making any additional changes to it. The `_eval()` function is ~250 lines long and contains about 17 `if` statements, sometimes nested 2-3 levels deep. As written, it is extremely difficult to make enhancements or fixes without breaking existing functionality.

**Comment by jpy-git** [comment]
I think this is fine to be made separate from L052. All this rule needs to do is detect if there are two or more consecutive semi-colons and delete the extra ones. L052 can handle the actual positioning

**Comment by jpy-git** [comment]
Going to pick this up as a separate rule

**Comment by annebelleo** [comment]
I believe this is now implemented by #7094
